content/blog/there-and-back-again/index.md

```diff
@@ -0,0 +1,505 @@
+toString suit =
+    case suit of
+        Green ->
+            "Green"
+
+        Yellow ->
+            "Yellow"
+
+        Red ->
+            "Red"
+
+        Blue ->
+            "Blue"
+```
+
+And use it to dry up the view:
+
+```elm
+selectOptions : List (Html Msg)
+selectOptions =
+    List.map
+        (\color ->
+            Html.option
+                [ Attrs.value (toString color) ]
+                [ text (toString color) ]
+        )
+        [ Green, Yellow, Red, Blue ]
+
+view : Model -> Html Msg
+view model =
+    div []
+        [ Html.h2 [] [ text "Choose your Power Ranger Suit" ]
+        , Html.select []
+            (Html.option [ Attrs.value "" ] [ text "Choose" ]
+                :: selectOptions
+            )
+        ]
+```
+
+You can play around with this [Ellie app](https://ellie-app.com/cDq7YCwvfpBa1)
+to see how that looks like.
+
+If you're a bit more experienced with Elm, you will notice that you can
+choose an option in the select, but that won't update the `Model`.
+Let's prove that assumption by changing the view to:
+
+```elm
+view : Model -> Html Msg
+view model =
+    div []
+        [ Html.h2 [] [ text "Choose your Power Ranger Suit" ]
+        , Html.select []
+            (Html.option [ Attrs.value "" ] [ text "Choose" ]
+                :: selectOptions
+            )
+        , Html.p []
+            [ case model.chosenSuit of
+                Nothing ->
+                    text "No suit was chosen."
+
+                Just suit ->
+                    text ("Your choice: " ++ toString suit)
+            ]
+        ]
+```
+
+As you can see, no matter how you change the view, the value won't be
+updated. This happens because we haven't defined a `onInput` event on the
+select :(
+
+```elm
+import Html.Events as Events
+
+view : Model -> Html Msg
+view model =
+    div []
+        [ Html.h2 [] [ text "Choose your Power Ranger Suit" ]
+        , Html.select [ Events.onInput ChooseSuit ]
+            (Html.option [ Attrs.value "" ] [ text "Choose" ]
+                :: selectOptions
+            )
+        , Html.p []
+            [ case model.chosenSuit of
+                Nothing ->
+                    text "No suit was chosen."
+
+                Just suit ->
+                    text ("Your choice: " ++ toString suit)
+            ]
+        ]
+
+type Msg
+    = ChooseSuit String
+
+update : Msg -> Model -> Model
+update msg model =
+    case msg of
+        ChooseSuit value ->
+            { model | chosenSuit = fromString value }
+```
+
+What's that `fromString` function? Here it is:
+
+```elm
+fromString : String -> Maybe Suit
+fromString str =
+    case str of
+        "Green" ->
+            Just Green
+
+        "Yellow" ->
+            Just Yellow
+
+        "Red" ->
+            Just Red
+
+        "Blue" ->
+            Just Blue
+
+        _ ->
+            Nothing
+```
+
+In this [Ellie app](https://ellie-app.com/cDqjV9LLFX6a1), we can see
+that everything is working nicely :)
```

Comment:
FWIW, I think everything up to this point could be condensed a bit. My thinking:
  - Until now, the text was fun to follow but the motivation wasn't clear to me
  - I'm guessing that anyone that's likely to appreciate the problem and try out elm-review to work around that will probably have seen many basic TEA examples already and doesn't need that much hand holding

Reply:
Yup, good point, I've thought about doing that before.